instrumentation/que/lib/opentelemetry/instrumentation/que/patches/que_job.rb

```diff
@@ -43,27 +43,35 @@ def enqueue(*args, job_options: {}, **arg_opts)
                   OpenTelemetry.propagation.inject(tags, setter: TagSetter)
                 end
 
-                job_options = job_options.merge(tags: tags)
-                job = super(*args, job_options: job_options, **arg_opts)
+                # In Que version 2.1.0 `bulk_enqueue` was introduced and in order
+                # for it to work, we must pass `job_options` to `bulk_enqueue` instead of enqueue.
+                if Gem.loaded_specs['que'].version >= Gem::Version.new('2.1.0') && Thread.current[:que_jobs_to_bulk_insert]
```

Comment:
Is checking a thread-local var the only way to do it? I would feel better about it if we could use something the `Que` provides to manipulate this rather than manipulating these ourselves (feels like we'll have compatibility problems with this down the line if we aren't careful).

(Also: is there a way to not check `Gem.loaded_specs` every time? I feel like we could / should memoize that at least, feels like a lot more work than we really want to do if we can avoid it. Not like it's going to change on us while it's running! At least ... I hope not 😅 )

Reply:
The thread-local variable is modified without using any function within `enqueue` and `bulk_enqueue` methods and they don't provide any way of doing it through `Que` as well so I don't think there is a _good_ way to do it here as well unless they update it in the future.

Also, I added memoization to the gem version to not call `Gem.loaded_specs` every time.